### PR TITLE
Moved winlogbeat pipeline to central pipeline

### PR DESCRIPTION
**Pull request by oakrizan**
## Proposed commit message
In scope of https://github.com/elastic/ingest-dev/issues/3072 - moved winlogbeatt pipeline triggering to central pipeline
BK Builds: https://buildkite.com/elastic/beats/builds?branch=oakrizan%3Acentralised-pipeline-winlogbeat

Xpack/Heartbeat failure unrelated to current PR, related issue https://github.com/elastic/ingest-dev/issues/3196

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: .buildkite/scripts/common.sh [hunk 1]
```diff
@@ -72,10 +71,6 @@ xpack_packetbeat_changeset=(
   "^x-pack/packetbeat/.*"
   )
 
-xpack_winlogbeat_changeset=(
```
Comment (dliappis) [inline .buildkite/scripts/common.sh:75]: this is unrelated to this PR (this PR is about `winlogbeat`)

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml [hunk 2]
```diff
@@ -2,48 +2,149 @@
...
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "winlogbeat/build/*.xml"
+          - "winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "Winlogbeat: Win-2022 Unit Tests"
+
+  - group: "Extended Windows Tests"
```
Comment (dliappis) [inline .buildkite/winlogbeat/pipeline.winlogbeat.yml:92]:
```suggestion
  - group: "Winlogbeat Extended Windows Tests"
```

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -285,3 +285,29 @@ steps:
...
+          watch:
+            - path:
+                - winlogbeat/
+                - .buildkite/winlogbeat/pipeline.winlogbeat.yml
+                - .buildkite/scripts
+                - .buildkite/hooks
+                #OSS
+                - go.mod
+                - pytest.ini
+                - dev-tools/**
```
Comment (pazone) [inline .buildkite/pipeline.yml:302]: I'd recommend to remove `**`

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml [hunk 4]
```diff
@@ -2,48 +2,149 @@
...
+    key: "winlogbeat-extended-win-tests"
+
```
Comment (dliappis) [inline .buildkite/winlogbeat/pipeline.winlogbeat.yml:94]: should we add:

`if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*[Ww]indows.*/`


to allow running with the right label this extended group (see https://github.com/elastic/beats/blob/main/.buildkite/metricbeat/pipeline.yml#L158C5-L158C108)

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml [hunk 5]
```diff
@@ -2,48 +2,149 @@
...
+    steps:
+      - label: ":windows: Winlogbeat Wind-10 Unit Tests"
```
Comment (dliappis) [inline .buildkite/winlogbeat/pipeline.winlogbeat.yml:96]:
```suggestion
      - label: ":windows: Winlogbeat Win-10 Unit Tests"
```

File: .buildkite/scripts/common.sh (hunk 1 above)
Comment (pazone) [inline .buildkite/scripts/common.sh:75]: That was `xpack/winlogonbeat`. `winlogbeat_changeset` should be removed instead, right?

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml [hunk 6]
```diff
@@ -2,48 +2,149 @@
...
-
-  - label: ":linux: Load dynamic winlogbeat pipeline"
-    key: "winlogbeat-pipeline"
-    command: ".buildkite/scripts/generate_winlogbeat_pipeline.sh"
-    agents:
-      image: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci:latest"
-    notify:
-      - github_commit_status:
-          context: "${BEATS_PROJECT_NAME}: Load dynamic pipeline's steps"
+    depends_on:
```
Comment (dliappis) [inline .buildkite/winlogbeat/pipeline.winlogbeat.yml:133]:
```
    if: build.env("BUILDKITE_PULL_REQUEST") != "false"
    depends_on: "winlogbeat-mandatory-tests"
```

see https://github.com/elastic/beats/blob/main/.buildkite/metricbeat/pipeline.yml#L256C1-L257C45

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml [hunk 7]
```diff
@@ -2,48 +2,149 @@
 name: "beats-winlogbeat"
 
 env:
-  BEATS_PROJECT_NAME: "winlogbeat"
```
Comment (pazone) [inline .buildkite/winlogbeat/pipeline.winlogbeat.yml:5]: Question: are we intentionally removing `BEATS_PROJECT_NAME`? It is used in various scripts

**Review by pazone — CHANGES_REQUESTED** [review]

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml [hunk 8]
```diff
@@ -2,48 +2,149 @@
...
-          context: "${BEATS_PROJECT_NAME}: Load dynamic pipeline's steps"
+    depends_on:
+      - step: "winlogbeat-mandatory-tests"
+
+  - group: "Winlogbeat Packaging"
+    key: "winlogbeat-packaging"
+    steps:
+      - label: ":ubuntu: Winlogbeat Packaging Ubuntu x86_64"
+        key: "packaging-linux"
+        command: "cd winlogbeat && mage package"
```
Comment (dliappis) [inline .buildkite/winlogbeat/pipeline.winlogbeat.yml:141]:
```suggestion
        command: |
          cd winlogbeat
          mage package
```

**Review by dliappis — CHANGES_REQUESTED** [review]
Thanks! Left a few comments.

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml (hunk 7 above)
Comment (dliappis) [reply .buildkite/winlogbeat/pipeline.winlogbeat.yml:5]: > Question: are we intentionally removing `BEATS_PROJECT_NAME`? It is used in various scripts

we yes want to retire it with an aim to remove as many scripts as possible in the near future.

Comment (oakrizan) [reply .buildkite/winlogbeat/pipeline.winlogbeat.yml:5]: Yep, that was asked during previous PRs.

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml [hunk 9]
```diff
@@ -2,48 +2,152 @@
...
+        artifact_paths:
+          - "winlogbeat/build/*.xml"
+          - "winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "Winlogbeat: Win-2022 Unit Tests"
+
+  - group: "Winlogbeat Extended Windows Tests"
+    key: "winlogbeat-extended-win-tests"
+    if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*[Ww]indows.*/
```
Comment (dliappis) [inline .buildkite/winlogbeat/pipeline.winlogbeat.yml:94]: Can we also trigger a build after setting the GH label e.g. `:Windows` to see if the extended tests also pass?

**Review by dliappis — APPROVED** [review]
LGTM provided that extended windows tests also pass (see comment)

**Review by pazone — APPROVED** [review]
